operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -38,6 +38,10 @@ type ImageContentSourcePolicySpec struct {
 	// (e.g. `a, b` vs. `b, a`) the configuration is not rejected but the resulting order is unspecified.
 	// +optional
 	RepositoryDigestMirrors []RepositoryDigestMirrors `json:"repositoryDigestMirrors"`
+	// repositoryMirrors is the same as repositoryMirrors, allows images referenced by image digests in pods to be
+	// pulled from alternative mirrored repository locations. Provides AllowMirrorByTags to enable to
+	// use mirrors without specifying image digests.
+	RepositoryMirrors []RepositoryMirrors `json:"repositoryMirrors"`
```

Comment:
omitempty as optional (specify!) slice.

Reply:
Why is this thread marked resolved?  I don't see either `+optional` or `omitempty` on this property yet.

Reply:
@wking @sttts  https://github.com/openshift/api/blob/b63d499a70cef2320805dd7c9e12c4e06a95b9c4/operator/v1alpha1/types_image_content_source_policy.go#L40 does not set omitempty to keep the same implementation as above.  Did we not set omitempty for some reasons?

Reply:
Still unresolved. I would expect omitempty here.

Reply:
And also list-type is missing to be specified.

Reply:
@sttts Could you elaborate with some example? what is the list-type need to be added?

Reply:
@QiWang19 A list type in kubernetes is explicit... It sorta looks like this: https://github.com/openshift/api/commit/81e1a0ee7f24a8bd826512d51757c0cc0e8d6dc8

Reply:
Is it adding comment `// +listType=set`to specify the RepositoryMirrors can't contain duplicate items? like https://github.com/openshift/api/pull/628/files#diff-a0aeadd18ed80a3b03f604a43c6465c5b1c5eb2106bf85e4ce9a984ef9a1c157R122.
Compared with the `RepositoryDigestMirrors`, we only need add new field `allowMirrorByTags`. Is it necessary convert it to kubernetes list type?

Reply:
I don't think set makes sense. What is the primary key of this field?

Reply:
```
// +listType=map
// +listMapKey=source
```
fixed with map,the source should be an identify key.

Reply:
This is done.